packages/flutter/lib/src/services/message_codecs.dart

```diff
@@ -271,26 +274,23 @@ class StandardMessageCodec implements MessageCodec<dynamic> {
     if (message == null)
       return null;
     final ReadBuffer buffer = new ReadBuffer(message);
-    final dynamic result = _readValue(buffer);
+    final dynamic result = readValue(buffer);
     if (buffer.hasRemaining)
       throw const FormatException('Message corrupted');
     return result;
   }
 
-  static void _writeSize(WriteBuffer buffer, int value) {
-    assert(0 <= value && value <= 0xffffffff);
-    if (value < 254) {
-      buffer.putUint8(value);
-    } else if (value <= 0xffff) {
-      buffer.putUint8(254);
-      buffer.putUint16(value);
-    } else {
-      buffer.putUint8(255);
-      buffer.putUint32(value);
-    }
-  }
-
-  static void _writeValue(WriteBuffer buffer, dynamic value) {
+  /// Writes [value] to [buffer] by first writing a type discriminator
+  /// byte, then the value itself.
+  ///
+  /// Type discriminators 0 through 127 inclusive are reserved for use by the
+  /// base class.
+  ///
+  /// The codec can be extended by overriding this method, calling super
+  /// for values that the extension does not handle. Type discriminators
+  /// used by extensions must be greater than or equal to 128 in order to avoid
+  /// clashes with any later extensions to the base class.
+  void writeValue(WriteBuffer buffer, dynamic value) {
```

Comment:
maybe mention that it's ok to call this recursively, e.g. to implement writing a list of values.